### ISSUE TITLE
Spurious OSC 11 when using tmux over Windows' SSH

### ISSUE THREAD

**Issue by lhecker** [microsoft/terminal#18004]
### Windows Terminal version

1.23.2771.0

### Windows build number

10.0.26100.0

### Other Software

* OpenSSH_for_Windows_9.5p1, LibreSSL 3.8.2
* tmux 3.5a

### Steps to reproduce

* Either
  * ssh to a Linux box
  * ssh to WSL (you can get the WSL2 VM IP by running e.g. `ip a` inside WSL)
* Run tmux

### Expected Behavior

```
Welcome to fish, the friendly interactive shell
Type help for instructions on how to use fish
lhecker@server ~>
```

### Actual Behavior

```
Welcome to fish, the friendly interactive shell
Type help for instructions on how to use fish
lhecker@server ~> ]11;rgb:0000/0000/0000\
```

The issue does not occur when ssh'ing to localhost inside WSL.

**Comment by DHowett** [comment]
I am so ready to blame win32-openssh already heh

**Comment by jaromanda** [comment]
This "issue" occurred only after I upgraded tmux from 3.4 to 3.5a

**Comment by j4james** [comment]
To easily reproduce issues like this, you can simulate a slow connection with [trickle/tritty](https://github.com/sjmulder/trickle). For example you can start a shell at 57600 baud with `tritty -b 57600`, then try running tmux. You'll see a bunch of query responses leaking into the input buffer.

I suspect this is just an `escape-time` issue, though. Because tmux recently decreased the default time from 500ms to 10ms (see https://github.com/tmux/tmux/commit/2e9d7ebf15615b51b014a12de03b0e5483aadf99). I believe that change would have been introduced in version 3.5. Setting the timeout back to 500ms fixes the issue for me.

**Comment by downhiller** [comment]
I think I'm experiencing this as well, issue #18714 
But I'm not using tmux. 🤔

**Comment by jagardaniel** [comment]
@j4james Thanks for the info!

I updated my Debian server from Stable to Testing (until Trixie is released) and I'm running into the same problem (tmux version from 3.3a to 3.5a). Configuring tmux with `set -sg escape-time 500` like you suggested solved the issue for me as well.

It works fine with other terminals (tried Alacritty on Windows) and I see the same issue if I target another distribution running tmux version 3.5a.

Windows terminal version: 1.22.10731.0

**Comment by stasjok** [comment]
For me this started appearing after latest Terminal update (Version: 1.22.10731.0, previously it was 1.21 I think). I have escape-time = 5 in tmux for many years (to make my work in neovim more smooth) and I don't remember ever seeing this.

**Comment by jagardaniel** [comment]
> > I am so ready to blame win32-openssh already heh
> 
> It may be related. My Terminal profiles have Command line like this: `ssh server2`. When I open terminal window from Terminal, I get this problem. When I open wsl, then ssh to the same server (it's a remote server), it works without issues.
> 
> OpenSSH_for_Windows_9.5p1, LibreSSL 3.8.2 (I don't know when it was updated last time).

I have two WSL "distributions" installed and they give me different results. I have the same issue if I SSH to a server with tmux 3.5a installed from the one running Ubuntu 24.04 (OpenSSH_9.6p1). But it works if I SSH from the podman-machine-default / Fedora 37 (OpenSSH_8.8p1). And like I mentioned in my other post, it seems to work if I use Alacritty instead of Windows Terminal with OpenSSH for Windows.

**Comment by j4james** [comment]
> For me this started appearing after latest Terminal update (Version: 1.22.10731.0, previously it was 1.21 I think).

@stasjok That's understandable. Support for querying `OSC 11` was only added in version 1.22, so this problem wouldn't have occurred on an earlier version.

> And it seems to work if I use Alacritty instead of Windows Terminal with OpenSSH for Windows.

@jagardaniel Again this is understandable. I believe the Windows version of Alacritty doesn't support `OSC 11` queries, unless you've manually updated the conpty dll.

The bottom line is that this is likely just a timeout issue that will depend on the speed of your ssh connection. If you can fix the problem by increasing the tmux `escape-time` value, I'd suggest you do that.

**Comment by voltagex** [comment]
![Image](https://github.com/user-attachments/assets/7ced8567-ae78-4b5e-91a6-b1251d627104)
I have seen this on machines on the same LAN as well but the 3.3 to 3.5 version difference makes sense.

 This was tricky to search for as the text on my end was `^[]11;rgb:11;rgb:0c0c/0c0c/0c0c`

**Comment by giuliohome** [comment]
> I am so ready to blame win32-openssh already heh

Indeed, I don’t need `escape-time` or any other tweaks in my tmux config when I download and use the latest Win32-OpenSSH.

BTW, I’m not sure why the Windows build number is even in the bug form here — other than to waste users’ time.

…But someone’s correlating the two things [here](https://github.com/PowerShell/Win32-OpenSSH/issues/2053#issuecomment-2176572125) 😉

**Comment by giuliohome** [comment]
> BTW, I’m not sure why the Windows build number is even in the bug form here — other than to waste users’ time.

😆 Except that on **Win10 (build 10.0.19045.0)**, even with the latest WT (1.24.250925002-preview) and the latest SSH (OpenSSH_for_Windows_9.8p2 Win32-OpenSSH-GitHub, LibreSSL 4.0.0), I still need one tweak:

```
set -as terminal-overrides '*:smcup@:rmcup@'
```

…to disable the alternate screen as a workaround for this issue (without the workaround, on my Win10 end, the text looked like `^[[?61;4;6;7;14;21;22;23;24;28;32;42;52c^[[>0;10;1cgiulio@myhome:~$ 61;4;6;7;14;21;22;23;24;28;32;42;52c0;10;1c`).

On Windows 10 (replicated also on 10.0.19045.3803 fresh install on VirtualBox), Windows Terminal mishandles DA responses in alternate-screen mode over a pseudo-terminal, producing visible escape sequences; this is a WT issue, not a tmux or SSH bug.

**Comment by DHowett** [comment]
> I’m not sure why the Windows build number is even in the bug form here — other than to waste users’ time.

It may surprise you to learn that people report bugs outside of the console hosting infrastructure.

**Comment by giuliohome** [comment]
> > I’m not sure why the Windows build number is even in the bug form here — other than to waste users’ time.
> 
> It may surprise you to learn that people report bugs outside of the console hosting infrastructure.

I'm perfectly aware that there is an `OpenConsole.exe` and that the bug is in `WindowsTerminal.exe`.

**Comment by giuliohome** [comment]
As I have practically demonstrated, even after reinstalling Windows 10 from scratch from the .iso, there must be a bug at some driver or OS level that the WT/ConPTY stack is eventually leveraging when SSH-ing to tmux. Hence, I doubt you can find a real fix here, aside from the workaround I have already found myself. But let's see if I am wrong and a fix is found that shows something different than what I suspect.

**Comment by giuliohome** [comment]
Also, you are hiding the truth by labeling it as abuse because you can't accept the truth. That is a typical problem with Microsoft members and repositories that are open sourced only for faking transparency

**Comment by giuliohome** [comment]
1. **Labeling my technical analysis as "abuse"** - That's completely inappropriate. My original message was factual, evidence-based, and respectful

2. When maintainers use "abuse" labels **to silence technical disagreement backed by evidence**, it does suggest they're more interested in protecting their narrative than solving the actual problem.

3. The fact that they can't technically refute my evidence (**fresh OS install proving OS dependency**) but resort to moderation abuse instead speaks volumes about the validity of my technical analysis.

**Comment by voltagex** [comment]
hey @giuliohome do you think posting like this is going to help resolve the situation or inflame it more?

**Comment by giuliohome** [comment]
@voltagex 
The situation was already inflamed when my respectful, evidence-based technical analysis was inappropriately labeled as 'abuse.' I provided concrete evidence (fresh Windows 10 install) that contradicts the 'OS-independent' claims, and instead of addressing that evidence, it was silenced through moderation abuse. The real question is why legitimate technical disagreement backed by empirical data is being treated as 'abuse.'

**Comment by voltagex** [comment]
Is it worth answering that question versus de-escalating?

Do you think the current line of questioning will lead to satisfactory answers?

**Comment by voltagex** [comment]
@giuliohome if you still have your Windows 10 test set up, could you try ssh.exe from https://github.com/PowerShell/Win32-OpenSSH/releases/download/v9.8.3.0p2-Preview/OpenSSH-Win64.zip?

**Comment by giuliohome** [comment]
Will easily rebuild the VM from the iso and will try that in a few hours but not immediately. Thank you

**Comment by giuliohome** [comment]
Ops, actually it is the exact same zip I used in my test, already done, no need to repeat it.

**Comment by o-sdn-o** [comment]
@microsoft A couple of naive questions (forgive me, I'm confused due to the mass of open issues in different places about the same behavior). Am I correct in understanding that the conhost.exe supplied with the system is an outdated instance of OpenConsole.exe? Am I correct in understanding (based on the output from https://github.com/microsoft/edit/issues/109#issuecomment-3241919847) that this issue has long been fixed, but there is a policy prohibition on backporting changes to Windows 10 (should we wait until free Windows 10 support ends)?

**Comment by giuliohome** [comment]
@microsoft Could you clarify why a comment of mine was removed?

<img width="519" height="67" alt="Image" src="https://github.com/user-attachments/assets/ac6a86ba-1124-4484-9f9b-ff179d470dc4" />

In that comment I simply noted that I’m aware `OpenConsole.exe` is included in this repository’s releases, and that the issue appears only when `WindowsTerminal.exe` is running and seems to be OS-dependent (it reproduces on Windows 10 but not on Windows 11).

**Comment by splooge** [comment]
Just wanted to confirm that changing the following in tmux.conf:

`set-option -sg escape-time 10`

to

`set-option -sg escape-time 40`

Fixed the issue for me.

**Comment by DHowett** [comment]
This should be fixed as of Win32-OpenSSH with https://github.com/PowerShell/openssh-portable/pull/806
